### DISCUSSION TITLE
RXMATH Plugin added

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#389] [Show and tell]
I added the plugin **RXMATH** containing some mathematical functions based on the BREXX math functions:

- acos() 
- asin() 
- atan() 
- cos () 
- cosh() 
- exp () 
- log () 
- log10() 
- pow10() 
- sin () 
- sinh() 
- sqrt() 
- tan () 
- tanh()

If something is missing let know.

**Comment by rvjansen** [comment]
very impressive!

**Reply by Peter-Jacob** [reply]
It was just the list, the functions itself are C functions anyway

**Comment by Peter-Jacob** [comment]
I integrated almost all of the standard C-math functions in the rxmath plugin

1. **Arithmetic Functions**
 ceil(x) Returns the smallest integer greater than or equal to x 
 floor(x) Returns the largest integer less than or equal to x
 fabs(x) Returns the absolute value of x 
 fmod( x,  y) Returns the remainder of x 
 round(x) Rounds x
 trunc(x) Truncates x 
2. **Power and Exponential Functions** 
 pow( x,  y) Computes (x raised to the power of y). 
 sqrt(x) Returns the square root of x
 cbrt(x) Returns the cube root of x
 exp(x) Returns the power of x to the natural logarithms.
 exp2(x) Computes base-2 exponential of x
 log(x) Computes the natural logarithm of x
 log10(x) Computes the base-10 logarithm of x
 log2(x) Computes the base-2 logarithm (
 hypot( x,  y) Returns the hypotenuse (sqrt(x**2,y**2))
3. **Trigonometric Functions**
 sin(x) Computes the sine of x
 cos(x) Computes the cosine of x
 tan(x) Computes the tangent of x
 asin(x) Computes the arcsine of x
 acos(x) Computes the arccosine of x
 atan(x) Computes the arctangent of x
4. **Hyperbolic Functions**
 sinh(x) Computes the hyperbolic sine of x
 cosh(x) Computes the hyperbolic cosine of x
 tanh(x) Computes the hyperbolic tangent of x
 asinh(x) Computes the hyperbolic arcsine of x
 acosh(x) Computes the hyperbolic arccosine of x
 atanh(x) Computes the hyperbolic arctangent of x
5. **Error and Gamma Functions**
 erf(x) Computes the error function of x
 erfc(x) Computes the complementary error function of x
 tgamma(x) Computes the gamma function of 
 lgamma(x) Computes the natural logarithm of the absolute value of the gamma function.
6. **Statistical Functions**
mean(array)  mean of a float array 
stddev(array) standard deviation of an array 
covar(array1,array2)  covariance factor of 2 arrays 
correl(array1,array2)  correlation factor of 2 arrays 
7. **Other Functions**
  djb2("any-string")  creates a hash of the string  (DJB2-funktion by Daniel J. Bernstein)
   say djb2("The quick brown fox jumps over the lazy dog") is 885799134
  murmur("any-string",seed)  creates a hash by MURMUR algorithm   
  fnv1a("any-string")  creates  FNV1A hash
  crc32("any-string")  creates  CRC32 hash
8. **Constants**  
PI() returns pi
Euler() return the Euler constand
